File: Java/Etc/암호.java
import java.util.ArrayList;
import java.util.Scanner;

public class Main {

    public String solution(int n, String word) {
        String answer = "";

        for (int i = 0 ; i < n ; i++) {
            String temp = word.substring(0, 7).replace('#', '1').replace('*', '0');
            int num = Integer.parseInt(temp, 2);
            answer += (char) num;
            word = word.substring(7);
        }

        return answer;
    }

    public static void main(String[] args) {
        Main T = new Main();
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        String word = scanner.next();

        System.out.println(T.solution(n, word));
    }

}

File: Python/Etc/암호.py
num = int(input())
word = input()
n = len(word) // num

data = []
temp = ''

word = word.replace('#','1')
word = word.replace('*', '0')

for i in range(0, len(word)-1, len(word)//num):
    data.append(word[i:i+len(word)//num])

result = []
for binary_str in data:
    result.append(int(binary_str, 2))

final = []
for n in result:
    final.append(chr(n))

print(''.join(final))
